### ISSUE TITLE
Regent: RDIR issue with privileges

### ISSUE THREAD

**Issue by rupanshusoi** [StanfordLegion/legion#1115]
The following

```
import "regent"

task foo(r: region(ispace(int1d), int),
         s: region(ispace(int1d), int),
         t: region(ispace(int1d), int))
where
  reads writes(s), reads(r, t) do
end

task main()
  var r = region(ispace(int1d, 10), int)
  var p = partition(equal, r, ispace(int1d, 5))

  for i in ispace(int1d, 5) do
    foo(p[i], p[i/2], p[i/3])
  end
end
regentlib.start(main)

```

fails with

`legion/language/src/rdir/plugin/src/regent/flow_from_ast.t:938: assertion failed`

Changing the privileges to `writes(r), reads(s, t)` works around the bug.

**Comment by elliottslaughter** [comment]
In general, RDIR does a poor job of handling conflicting privileges on aliased regions. In this case, it sees `p[i]` and `p[i/2]` as aliased and then runs into trouble because one is read and one is write. In some other cases this could be fixed by improving the tracking of disjointness in the compiler, but I think in this case the aliasing is fundamental.

Actually, I'd expect this test to fail without RDIR. When `i` is `0` then all the indices would compute to `0` and I'd expect the runtime to fail with an error, even if the compiler doesn't say anything.

**Comment by rupanshusoi** [comment]
> Actually, I'd expect this test to fail without RDIR. When `i` is `0` then all the indices would compute to `0` and I'd expect the runtime to fail with an error, even if the compiler doesn't say anything.

That's right, but we could change the loop parameters and still have the same problem. In any case, I thought opening an issue might help keep track of it.

**Comment by elliottslaughter** [comment]
Sure, RDIR isn't precise and you could find a set of index expressions that would make it work in the runtime and still break RDIR.

Yes, let's keep this open in case anyone else hits it.
